bot/plotting/comparison_generator.py

```diff
@@ -127,15 +130,36 @@ def comparison_generator(
 
                 is_experiment = False
 
+                # vary "Current function [A]" and "Ambient temperature [K]"
+                # if comparing models with a constant discharge
+                if number_of_comp != 1:
```

Comment:
I wonder if it would be easier to read (fewer if statements) to have one script for doing model comparisons and a separate one for comparing a single model with different parameters? There is probably some shared functionality you could put in utils or elsewhere

Reply:
The model_comparison could pick 2-3 models to compare and then do either
1. constant current discharge with a random "Current function [A]" and "Ambient temperature [K]
2. random experiment with random "Ambient temperature [K]

The parameter_comparison could then pick a single model and do 
1. constant current discharge with a random "Current function [A]" and "Ambient temperature [K]
2. random experiment with random "Ambient temperature [K]
and in both cases you vary a single parameter at random (in addition to varying the current and/or temperature)

Reply:
Then in `tweet_plot` you could make a random choice from `["model comparison", "parameter comparison", "degradation comparison"]` and call the appropriate function. 

Oh and in all of those cases you will also pick a random parameter set to start out with as you already do

Reply:
Yes, the `comparison_generator` is getting messier with every PR, I am working on refactoring the random part out of it, I'll push and create a PR for it soon. If it looks messy even after pushing that code then I will definitely split it into `"model comparison"` and `"parameter comparison"`.

Reply:
Sounds good!